orchestrator/graphql/schema.py

```diff
@@ -0,0 +1,146 @@
+StrawberryModelType = dict[str, StrawberryPydanticModel]
+
+GRAPHQL_MODELS: StrawberryModelType = {"ProductModelGraphql": ProductModelGraphql}
+
+
+@strawberry.federation.type(description="Orchestrator queries")
+class Query:
+    processes: Connection[ProcessType] = authenticated_field(
+        resolver=resolve_processes, description="Returns list of processes"
+    )
+    products: Connection[ProductType] = authenticated_field(
+        resolver=resolve_products, description="Returns list of products"
+    )
+    subscriptions: Connection[SubscriptionInterface] = authenticated_field(
+        resolver=resolve_subscriptions, description="Returns list of subscriptions"
+    )
+    settings: StatusType = authenticated_field(
+        resolver=resolve_settings,
+        description="Returns information about cache, workers, and global engine settings",
+    )
+
+
+Mutation = merge_types("Mutation", (SettingsMutation,))
+
+
+class OrchestratorGraphqlRouter(GraphQLRouter):
+    def __init__(self, schema: strawberry.federation.Schema, **kwargs: Any) -> None:
+        super().__init__(schema, **kwargs)
+
+    @staticmethod
+    def _format_graphql_error(error: GraphQLError) -> GraphQLFormattedError:
+        if isinstance(error.original_error, HTTPStatusError):
+            error.extensions["http_status_code"] = {  # type: ignore
+                f"{error.original_error.request.url}": error.original_error.response.status_code
+            }
+        return format_error(error)
+
+    async def process_result(self, request: Request, result: ExecutionResult) -> GraphQLHTTPResponse:
+        data: GraphQLHTTPResponse = {"data": result.data}
+
+        if result.errors:
+            data["errors"] = [self._format_graphql_error(error) for error in result.errors]
+
+        return data
+
+
+class OrchestratorSchema(strawberry.federation.Schema):
+    def process_errors(
+        self,
+        errors: list[GraphQLError],
+        execution_context: Union[ExecutionContext, None] = None,
+    ) -> None:
+        """Override error processing to reduce verbosity of the logging.
+
+        https://strawberry.rocks/docs/types/schema#handling-execution-errors
+        """
+        for error in errors:
+            if (
+                isinstance(error.original_error, HTTPStatusError)
+                and error.original_error.response.status_code == HTTPStatus.NOT_FOUND
+            ):
+                message = str(error.original_error).splitlines()[0]  # Strip "For more info"
+                StrawberryLogger.logger.debug(message)
+            else:
+                StrawberryLogger.error(error, execution_context)
+
+
+def custom_context_dependency(
+    get_current_user: Callable = Depends(get_oidc_user),  # noqa: B008
+    get_opa_decision: Callable = Depends(get_opa_security_graphql),  # noqa: B008
+) -> CustomContext:
+    return CustomContext(get_current_user=get_current_user, get_opa_decision=get_opa_decision)
```

Comment:
I realize I forgot to change this in the my oauth2lib PR (#298); but `CustomContext` has been moved to `oauth2_lib.strawberry.OauthContext`. Could you change that in this PR?

Reply:
yea, also good to change the `CustomContext` to `OrchestratorContext`